Adapt changes from ROS2 rolling to ros2-devel branch

Hello, 
The updates from upstream ROS2 causes diagnostics to have deprecation messages when compiling; 

```
--- stderr: diagnostic_updater
In file included from /opt/underlay_ws/src/ros/diagnostics/diagnostic_updater/src/example.cpp:35:
/opt/underlay_ws/src/ros/diagnostics/diagnostic_updater/include/diagnostic_updater/diagnostic_updater.hpp: In constructor ‘diagnostic_updater::Updater::Updater(std::shared_ptr<rclcpp::node_interfaces::NodeBaseInterface>, std::shared_ptr<rclcpp::node_interfaces::NodeLoggingInterface>, std::shared_ptr<rclcpp::node_interfaces::NodeParametersInterface>, std::shared_ptr<rclcpp::node_interfaces::NodeTimersInterface>, std::shared_ptr<rclcpp::node_interfaces::NodeTopicsInterface>, double)’:
/opt/underlay_ws/src/ros/diagnostics/diagnostic_updater/include/diagnostic_updater/diagnostic_updater.hpp:393:28: warning: ‘rclcpp::Duration::Duration(rcl_duration_value_t)’ is deprecated: Use Duration::from_nanoseconds instead or std::chrono_literals. For example:rclcpp::Duration::from_nanoseconds(int64_variable);rclcpp::Duration(0ns); [-Wdeprecated-declarations]
  393 |     warn_nohwid_done_(false)
```

Can we adapt to the chages so we dont see this deprecation messages ? 
Thank you for your time. 

unrelated reference for myself
https://github.com/ros-planning/navigation2/pull/2111

--- a/.github/workflows/lint.yaml
+++ b/.github/workflows/lint.yaml
@@ -3,43 +3,20 @@ on:
   pull_request:
 
 jobs:
-  ament_xmllint:
-    name: ament_xmllint
-    runs-on: ubuntu-20.04
-    steps:
-    - uses: actions/checkout@v1
-    - uses: ros-tooling/setup-ros@0.0.23
-    - uses: ros-tooling/action-ros-ci@0.0.17
-      with:
-        linter: xmllint
-        package-name: diagnostic_aggregator diagnostic_updater self_test
-
-  ament_lint_cpp:
+  ament_lint:
     name: ament_${{ matrix.linter }}
     runs-on: ubuntu-20.04
     strategy:
       fail-fast: false
       matrix:
-          linter: [cppcheck, cpplint, uncrustify]
+          linter: [cppcheck, cpplint, flake8, pep257, uncrustify, xmllint]
     steps:
     - uses: actions/checkout@v1
-    - uses: ros-tooling/setup-ros@0.0.23
-    - uses: ros-tooling/action-ros-ci@0.0.17
+    - uses: ros-tooling/setup-ros@v0.1
       with:
-        linter: ${{ matrix.linter }}
-        package-name: diagnostic_aggregator diagnostic_updater self_test
-
-  ament_lint_py:
-    name: ament_${{ matrix.linter }}
-    runs-on: ubuntu-20.04
-    strategy:
-      fail-fast: false
-      matrix:
-        linter: [flake8, pep257]
-    steps:
-    - uses: actions/checkout@v1
-    - uses: ros-tooling/setup-ros@0.0.23
-    - uses: ros-tooling/action-ros-ci@0.0.17
+        required-ros-distributions: rolling
+    - uses: ros-tooling/action-ros-lint@v0.1
       with:
+        distribution: rolling
         linter: ${{ matrix.linter }}
         package-name: diagnostic_aggregator diagnostic_updater self_test

--- a/.github/workflows/test.yaml
+++ b/.github/workflows/test.yaml
@@ -16,9 +16,12 @@ jobs:
       matrix:
         os: [ubuntu-20.04]
     steps:
-      - uses: ros-tooling/setup-ros@0.0.23
-      - uses: ros-tooling/action-ros-ci@0.0.17
+      - uses: ros-tooling/setup-ros@v0.1
         with:
+          required-ros-distributions: rolling
+      - uses: ros-tooling/action-ros-ci@v0.1
+        with:
+          target-ros2-distro: rolling
           package-name: diagnostic_aggregator diagnostic_updater self_test
           colcon-mixin-name: coverage-gcc
           colcon-mixin-repository: https://raw.githubusercontent.com/colcon/colcon-mixin-repository/master/index.yaml

--- a/diagnostic_aggregator/src/aggregator.cpp
+++ b/diagnostic_aggregator/src/aggregator.cpp
@@ -71,7 +71,7 @@ Aggregator::Aggregator()
   if (!n_->get_parameters("", parameters)) {
     RCLCPP_ERROR(logger_, "Couldn't retrieve parameters.");
   }
-  RCLCPP_DEBUG(logger_, "Retrieved %d parameter(s).", parameters.size());
+  RCLCPP_DEBUG(logger_, "Retrieved %zu parameter(s).", parameters.size());
 
   for (const auto & param : parameters) {
     if (param.first.compare("pub_rate") == 0) {

--- a/diagnostic_aggregator/src/analyzer_group.cpp
+++ b/diagnostic_aggregator/src/analyzer_group.cpp
@@ -74,7 +74,7 @@ bool AnalyzerGroup::init(
     return false;
   }
   RCLCPP_INFO(
-    logger_, "Retrieved %d parameter(s) for analyzer group with prefix '%s'.", parameters.size(),
+    logger_, "Retrieved %zu parameter(s) for analyzer group with prefix '%s'.", parameters.size(),
     breadcrumb_.c_str());
 
   std::string ns, an_type, an_path, an_breadcrumb;

--- a/diagnostic_aggregator/src/generic_analyzer.cpp
+++ b/diagnostic_aggregator/src/generic_analyzer.cpp
@@ -70,7 +70,7 @@ bool GenericAnalyzer::init(
     return false;
   }
   RCLCPP_DEBUG(
-    rclcpp::get_logger("GenericAnalyzer"), "Retrieved %d parameter(s) for prefix '%s'.",
+    rclcpp::get_logger("GenericAnalyzer"), "Retrieved %zu parameter(s) for prefix '%s'.",
     parameters.size(), breadcrumb_.c_str());
 
   double timeout = 5.0;
@@ -156,8 +156,8 @@ bool GenericAnalyzer::init(
   {
     RCLCPP_ERROR(
       rclcpp::get_logger("generic_analyzer"),
-      std::string("GenericAnalyzer ") +
-      "'%s' was not initialized with any way of checking diagnostics. Name: %s, namespace: %s",
+      "GenericAnalyzer '%s' was not initialized with any way of checking diagnostics."
+      "Name: %s, namespace: %s",
       nice_name_.c_str(), path.c_str(), n->get_namespace());
     return false;
   }

--- a/diagnostic_updater/include/diagnostic_updater/diagnostic_updater.hpp
+++ b/diagnostic_updater/include/diagnostic_updater/diagnostic_updater.hpp
@@ -384,7 +384,7 @@ public:
     base_interface_(base_interface),
     timers_interface_(timers_interface),
     clock_(std::make_shared<rclcpp::Clock>(RCL_ROS_TIME)),
-    period_(static_cast<rcl_duration_value_t>(period * 1e9)),
+    period_(rclcpp::Duration::from_nanoseconds(period * 1e9)),
     publisher_(
       rclcpp::create_publisher<diagnostic_msgs::msg::DiagnosticArray>(
         topics_interface, "/diagnostics", 1)),
@@ -394,7 +394,7 @@ public:
   {
     period = parameters_interface->declare_parameter(
       "diagnostic_updater.period", rclcpp::ParameterValue(period)).get<double>();
-    period_ = rclcpp::Duration(static_cast<rcl_duration_value_t>(period * 1e9));
+    period_ = rclcpp::Duration::from_nanoseconds(period * 1e9);
 
     reset_timer();
   }
@@ -414,19 +414,11 @@ public:
   }
 
   /**
-   * \brief Sets the period as a rcl_duration_value_t
-   */
-  void setPeriod(rcl_duration_value_t period)
-  {
-    setPeriod(rclcpp::Duration(period));
-  }
-
-  /**
    * \brief Sets the period given a value in seconds
    */
   void setPeriod(double period)
   {
-    setPeriod(static_cast<rcl_duration_value_t>(period * 1e9));
+    setPeriod(rclcpp::Duration::from_nanoseconds(period * 1e9));
   }
 
   /**
@@ -539,7 +531,7 @@ private:
         error_msg += " For devices that do not have a HW_ID, set this value to 'none'.";
         error_msg += " This warning only occurs once all diagnostics are OK.";
         error_msg += " It is okay to wait until the device is open before calling setHardwareID.";
-        RCLCPP_WARN(logger_, error_msg);
+        RCLCPP_WARN(logger_, "%s", error_msg.c_str());
         warn_nohwid_done_ = true;
       }
 

--- a/diagnostic_updater/include/diagnostic_updater/publisher.hpp
+++ b/diagnostic_updater/include/diagnostic_updater/publisher.hpp
@@ -165,7 +165,7 @@ public:
   {
     std::string error_msg = "tick(void) has been called on a TopicDiagnostic.";
     error_msg += " This is never correct. Use tick(rclcpp::Time &) instead.";
-    RCLCPP_FATAL(error_logger_, error_msg);
+    RCLCPP_FATAL(error_logger_, "%s", error_msg.c_str());
   }
 
   /**

--- a/self_test/src/run_selftest.cpp
+++ b/self_test/src/run_selftest.cpp
@@ -72,9 +72,9 @@ public:
           (result_out->passed ? "PASSED" : "FAILED"), result_out->id.c_str());
 
         // for (size_t i = 0; i < result_out->status.size(); i++) {
-        auto counter = 1;
+        auto counter = 1lu;
         for (const auto & status : result_out->status) {
-          RCLCPP_INFO(this->get_logger(), "%2zd) %s", counter++, status.name.c_str());
+          RCLCPP_INFO(this->get_logger(), "%zu) %s", counter++, status.name.c_str());
           if (status.level == 0) {
             RCLCPP_INFO(this->get_logger(), "\t%s", status.message.c_str());
           } else if (status.level == 1) {

